lib/python/ray/services.py

```diff
@@ -54,6 +61,31 @@ def new_port():
 def random_name():
   return str(random.randint(0, 99999999))
 
+def kill_process(p):
+  """Kill a process.
+
+  Args:
+    p: The process to kill.
+
+  Returns:
+    True if the process was killed successfully and false otherwise.
+  """
+  if p.poll() is not None: # process has already terminated
+    return True
+  if RUN_PHOTON_PROFILER or RUN_PLASMA_MANAGER_PROFILER or RUN_PLASMA_STORE_PROFILER:
+    os.kill(p.pid, signal.SIGINT) # Give process signal to write profiler data.
+    time.sleep(0.1) # Wait for profiling data to be written.
+  p.kill()
+  time.sleep(0.05) # is this necessary?
```

Comment:
you need to let the victim process get a chance to process its pending signals. So, minimally, would be good to yield. I think it could be sufficient to just do ``time.sleep(0)``